cdap-watchdog/src/main/java/co/cask/cdap/logging/write/LogCleanup.java

```diff
@@ -77,62 +78,154 @@ public LogCleanup(FileMetaDataManager fileMetaDataManager, RootLocationFactory r
 
     LOG.debug("Log retention duration = {} ms", retentionDurationMs);
   }
-  
+
   @Override
   public void run() {
     LOG.info("Running log cleanup...");
     try {
       long tillTime = System.currentTimeMillis() - retentionDurationMs;
       final SetMultimap<String, Location> parentDirs = HashMultimap.create();
       final Map<String, NamespaceId> namespacedLogBaseDirMap = new HashMap<>();
-      fileMetaDataManager.cleanMetaData(tillTime,
-                                        new FileMetaDataManager.DeleteCallback() {
-                                          @Override
-                                          public void handle(NamespaceId namespaceId, final Location location,
-                                                             final String namespacedLogBaseDir) {
-                                            try {
-                                              deleteLogFiles(parentDirs, namespaceId, namespacedLogBaseDir, location);
-                                              namespacedLogBaseDirMap.put(namespacedLogBaseDir, namespaceId);
-                                            } catch (Exception e) {
-                                              LOG.error("Got exception when deleting path {}", location, e);
-                                              throw Throwables.propagate(e);
-                                            }
-                                          }
-                                        });
-
-      // clean log files which does not have corresponding meta data
+
+      cleanLogFiles(tillTime, namespacedLogBaseDirMap, parentDirs, MAX_META_FILES_SCANNED);
+
       try {
+        // clean log files which does not have corresponding meta data
         cleanFilesWithoutMeta(tillTime, namespacedLogBaseDirMap, parentDirs,
                               MAX_DISK_FILES_SCANNED, MAX_META_FILES_SCANNED);
       } catch (Exception e) {
-        LOG.error("Got exception while cleaning up disk files without meta data", e);
+        LOG.warn("Got exception while cleaning up disk files without meta data", e);
       }
 
       // Delete any empty parent dirs
       for (final String namespacedLogBaseDir : parentDirs.keySet()) {
         // this ensures that we only do doAs which will make an RPC call only once for a namespace
         NamespaceId namespaceId = namespacedLogBaseDirMap.get(namespacedLogBaseDir);
-        impersonator.doAs(namespaceId, new Callable<Void>() {
-          @Override
-          public Void call() throws Exception {
-            Set<Location> locations = parentDirs.get(namespacedLogBaseDir);
-            for (Location location : locations) {
-              deleteEmptyDir(namespacedLogBaseDir, location);
+        try {
+          impersonator.doAs(namespaceId, new Callable<Void>() {
```

Comment:
We are not handling `NamespaceNotFoundException` as a special case because we do not want to do any specific operation when we catch that. Thats why we are only catching `Exception`.